Nav bar mobile

GeorgePrimentas opened the pull request:
## Naming Rules

Name your PR like this: #4 & #5-NavigationBar-George Primentas

## Description

Introduced a Navigation Bar into the Header

## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #4 & #5 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

GeorgePrimentas committed:
Import NavBar component into Header

GeorgePrimentas committed:
NavBar component for mobile version

AndrianaOS commented on client/src/components/NavBar/NavBar.jsx:
```diff
@@ -0,0 +1,71 @@
+          </button>
+        ) : (
+          <nav className="header-links">
+            <a href="#">Team Registration</a>
```
Maybe instead of using the anchor tags we should probably use React Router and Links so we dont have to refactor the code at a later stage. 
So set the anchor tags to a Link and we can get the routes at a later stage. So something like this.

              `<Link to="/" className="list-links">
                Home
              </Link>

              <Link to="/form" className="list-links">
                Team Registration
              </Link>

              <Link to="/search" className="list-links">
                Search
              </Link>`

GeorgePrimentas replied:
I was not aware of React Router's existence until last Saturday when Ali (volunteer) told me it's a must. I'm happy to explore it and I have implemented the changes to the code. It seems it's worth to learn it. Thank you for giving me the opportunity to experiment with it.

AndrianaOS commented on client/src/components/NavBar/NavBar.jsx:
```diff
@@ -0,0 +1,71 @@
+  // Update windowWidth state on window resize
+  useEffect(() => {
+    const handleResize = () => setWindowWidth(window.innerWidth);
+    window.addEventListener("resize", handleResize);
```
I dont know much about this window resize event listener so im going to assume its alright as i saw it working before

AndrianaOS commented on client/src/components/NavBar/NavBar.jsx:
```diff
@@ -0,0 +1,71 @@
+            <a href="#" onClick={() => setShowMenu(false)}>About</a>
+          </div>
+        )}
+         {showAbout && <div className="about-content">About Us - Some content here</div>} {/* Display the About component */}
```
Probably take out this bit here as i doubt this is what you had in mind when you click about link when in the desktop version. See attached image: 
<img width="1440" alt="Screenshot 2023-08-04 at 9 27 12 am" src="https://github.com/Gayle-Thompson-Igwebike/GOOD-PR-v1/assets/113118206/2d74393e-59e2-4e97-b125-1b7ede16102d">

GeorgePrimentas replied:
Totally agree. This is something I missed while combining the two tickets into one pull request. Thank you!

AndrianaOS reviewed (changes requested):
There a few things that i noticed while doing the review. I have made some comments but generally very good coding. Learnt some new things line the window resize event listener

GeorgePrimentas committed:
Navigation Bar with React Router employed

Swapped <a> links with React Router <Link>s and amended the rest of the code accordingly so that the results are the same as with previous commits.

AndrianaOS reviewed (approved):
All good to be merged. There is a function in there that is not being used. Delete or comment it out.

GeorgePrimentas committed:
Deleted unused function

Deleted handleAboutClick function
